gateway/dataimpl/server_v1/errorhandler.go

```diff
@@ -145,6 +145,29 @@ func (e ErrorHandler) NewBucketExistsStatus(baseErr error, bucketName string) *s
 	return st
 }
 
+func (e ErrorHandler) NewBucketInvalidArgStatus(baseErr error, msg string, bucketName string) *status.Status {
+	if msg == "" {
+		msg = "could not retrieve invalid argument details"
+	}
+	st := status.New(codes.InvalidArgument, msg)
+
+	var sErr *cbmgmtx.ServerInvalidArgError
+	if errors.As(baseErr, &sErr) {
+		st = status.New(codes.InvalidArgument, sErr.Error())
```

Comment:
Should we be crafting our own message here using the content of the error?

Reply:
I thought the plan was to propagate the error in `arguement : reason` form in the case of a ServerInvalidArgError, which  the implementation of Error does. I don't see the advantage of crafting our own message unless we are changing the format of the message based on the argument or reason.

Reply:
While you are correct, part of the exercise of why we went down the road of implementing ServerInvalidArgError was to ensure that CNG was 'mostly in control' of its own error handling.  Using .Error() means that (for instance) changing gocbcorex's message to provide additional debugging information would change CNGs message.

Reply:
Done!